lib/eunit/include/eunit.hrl

```diff
@@ -128,211 +118,24 @@
--else.
--define(assertNotEqual(Unexpected, Expr),
-	begin
-	((fun (__X) ->
-	    case (Expr) of
-		__X -> erlang:error({assertNotEqual_failed,
-				     [{module, ?MODULE},
-				      {line, ?LINE},
-				      {expression, (??Expr)},
-				      {value, __X}]});
-		_ -> ok
-	    end
-	  end)(Unexpected))
-	end).
--endif.
 -define(_assertNotEqual(Unexpected, Expr),
 	?_test(?assertNotEqual(Unexpected, Expr))).
-
-%% Note: Class and Term are patterns, and can not be used for value.
-%% Term can be a guarded pattern, but Class cannot.
--ifdef(NOASSERT).
--define(assertException(Class, Term, Expr), ok).
--else.
--define(assertException(Class, Term, Expr),
-	begin
-	((fun () ->
-	    try (Expr) of
-	        __V -> erlang:error({assertException_failed,
-				      [{module, ?MODULE},
-				       {line, ?LINE},
-				       {expression, (??Expr)},
-				       {pattern,
-					"{ "++(??Class)++" , "++(??Term)
-					++" , [...] }"},
-				       {unexpected_success, __V}]})
-	    catch
-		Class:Term -> ok;
-	        __C:__T ->
-		    erlang:error({assertException_failed,
-				  [{module, ?MODULE},
-				   {line, ?LINE},
-				   {expression, (??Expr)},
-				   {pattern,
-				    "{ "++(??Class)++" , "++(??Term)
-				    ++" , [...] }"},
-				   {unexpected_exception,
-				    {__C, __T,
-				     erlang:get_stacktrace()}}]})
-	    end
-	  end)())
-	end).
--endif.
-
--define(assertError(Term, Expr), ?assertException(error, Term, Expr)).
--define(assertExit(Term, Expr), ?assertException(exit, Term, Expr)).
--define(assertThrow(Term, Expr), ?assertException(throw, Term, Expr)).
-
 -define(_assertException(Class, Term, Expr),
 	?_test(?assertException(Class, Term, Expr))).
 -define(_assertError(Term, Expr), ?_assertException(error, Term, Expr)).
 -define(_assertExit(Term, Expr), ?_assertException(exit, Term, Expr)).
 -define(_assertThrow(Term, Expr), ?_assertException(throw, Term, Expr)).
-
-%% This is the inverse case of assertException, for convenience.
-%% Note: Class and Term are patterns, and can not be used for value.
-%% Both Class and Term can be guarded patterns.
--ifdef(NOASSERT).
--define(assertNotException(Class, Term, Expr), ok).
--else.
--define(assertNotException(Class, Term, Expr),
-	begin
-	((fun () ->
-	    try (Expr) of
-	        _ -> ok
-	    catch
-		__C:__T ->
-		    case __C of
-			Class ->
-			    case __T of
-				Term ->
-				    erlang:error({assertNotException_failed,
-						  [{module, ?MODULE},
-						   {line, ?LINE},
-						   {expression, (??Expr)},
-						   {pattern,
-						    "{ "++(??Class)++" , "
-						    ++(??Term)++" , [...] }"},
-						   {unexpected_exception,
-						    {__C, __T,
-						     erlang:get_stacktrace()
-						    }}]});
-				_ -> ok
-			    end;
-			_ -> ok
-		    end
-	    end
-	  end)())
-	end).
--endif.
 -define(_assertNotException(Class, Term, Expr),
 	?_test(?assertNotException(Class, Term, Expr))).
+-define(_assertReceive(Guard, Expr), ?_test(?assertReceive(Guard, Expr))).
```

Comment:
@richcarl, was this line added due to #616? We reverted that patch after your comment there, at May 27. So I'm wondering if we should add it again, or remove this line? It created some confusion: http://bugs.erlang.org/browse/ERL-45?jql=

@saleyn